File: app/src/main/java/com/example/riven_chris/demos/MainActivity.java
package com.example.riven_chris.demos;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.os.CountDownTimer;
import android.os.Handler;
import android.view.View;
import android.webkit.WebChromeClient;
import android.webkit.WebSettings;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.bumptech.glide.Glide;
import com.bumptech.glide.GlideBuilder;
import com.facebook.common.util.UriUtil;
import com.facebook.drawee.backends.pipeline.Fresco;
import com.facebook.drawee.interfaces.DraweeController;
import com.facebook.drawee.view.SimpleDraweeView;
import com.facebook.imagepipeline.common.RotationOptions;
import com.facebook.imagepipeline.request.ImageRequest;
import com.facebook.imagepipeline.request.ImageRequestBuilder;

public class MainActivity extends AppCompatActivity {

    TextView tv;
    WebView webView;
    GradientTextView gtv;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Fresco.initialize(this);
//        Glide.init(this, new GlideBuilder());
        setContentView(R.layout.activity_main);
        webView = (WebView) findViewById(R.id.web_view);
        gtv = (GradientTextView) findViewById(R.id.gtv);
        gtv.onTime(0, "现在还剩下100%", false);
        final CountDownTimer countDownTimer = new CountDownTimer(50000, 1000) {
            @Override
            public void onTick(long millisUntilFinished) {
                float percent = millisUntilFinished / 50000.f;
                String text = "现在还剩下" + (int) (percent * 100) + "%了";
                gtv.onTime(1 - percent, text, true);
            }

            @Override
            public void onFinish() {
                gtv.onTime(0, "现在玩完啦", false);
            }
        };

        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                countDownTimer.start();
            }
        }, 5000);

        gtv.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, SecondActivity.class);
                startActivity(intent);
            }
        });

//        ActivityManager manager = (ActivityManager) getSystemService(Context.ACTIVITY_SERVICE);
//        Log.d("TAG", "memory: " + manager.getMemoryClass() + " runtime: " + (Runtime.getRuntime().maxMemory() / (1024 * 1024)));
//
//        tv = (TextView) findViewById(R.id.tv);
//        tv.setOnClickListener(new View.OnClickListener() {
//            @Override
//            public void onClick(View v) {
//                Intent intent1 = new Intent(MainActivity.this, SecondActivity.class);
//                Intent intent2 = new Intent(MainActivity.this, ThirdActivity.class);
//                startActivities(new Intent[]{intent1, intent2});
//            }
//        });
//
//        Fresco.getImagePipeline();
//        OkHttpClient okHttpClient = null;
//        okHttpClient.cancel("");

        WebSettings settings = webView.getSettings();
        settings.setJavaScriptEnabled(true);//开启对JavaScript的支持
        settings.setDefaultTextEncodingName("UTF-8");//设置字符编码

        settings.setSupportZoom(true);
        // 开启alert
        settings.setJavaScriptCanOpenWindowsAutomatically(true);
        // 开启按钮按下显示
        settings.setLightTouchEnabled(true);
        settings.setUseWideViewPort(true);
        webView.setWebChromeClient(new WebChromeClient());
        webView.setWebViewClient(new WebViewClient() {

        });
        String url = "http://paibot.fe.ptdev.cn/browser/?uid=6079814&token=bee5697b65194d9eadd042f0441b60ac&appid=1109&cid={\"uidlist\":[{\"name\":\"11\",\"uid\":\"6083504\",\"avatar\":\"\"},{\"name\":\"22\",\"uid\":\"6083692\",\"avatar\":\"\"}]}#!/loading?type=applicationmanage";
        webView.loadUrl(url);
    }
}
 

File: windows/src/main/java/com/riven_chris/windows/MainActivity.kt
package com.riven_chris.windows

import android.app.Dialog
import android.content.Context
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.view.ViewGroup
import android.view.ViewManager
import android.view.WindowManager
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun attachBaseContext(newBase: Context?) {
        super.attachBaseContext(newBase)
        val windowManger = newBase?.getSystemService(Context.WINDOW_SERVICE) as WindowManager
        Log.d("windowTest", "MainActivity attachBaseContext: ${windowManger}")
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        Log.d("windowTest", "decorView: ${window.peekDecorView()}")
        super.onCreate(savedInstanceState)
        Log.d("windowTest", "decorView: ${window.peekDecorView()}")
        setContentView(R.layout.activity_main)
        Log.d("windowTest", "decorView: ${window.peekDecorView()}")


        tv_second.setOnClickListener{
            startActivity(Intent(this, SecondActivity::class.java))
        }
    }

    fun test() {
        window
        window.setWindowManager(null, null, null)
        val windowManager: WindowManager = windowManager
        window.decorView
        val viewManager: ViewManager = getSystemService(Context.WINDOW_SERVICE) as WindowManager
        val view: View
        val dialog: Dialog
        val toast: Toast
        val viewGroup: ViewGroup

        setVisible(true)
    }


    override fun onContentChanged() {
        super.onContentChanged()
    }
}
